src/main/java/com/javadiscord/jdi/internal/api/DiscordRequest.java

```diff
@@ -0,0 +1,98 @@
+package com.javadiscord.jdi.internal.api;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpRequest.Builder;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.javadiscord.jdi.internal.utils.UrlUtils;
+
+public class DiscordRequest {
+    public enum Method {
+        GET, POST, DELETE, PUT;
+    }
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private Method method;
+    private String url = "https://discord.com/api";
```

Comment:
this can be a constant

Reply:
not necessary so i didn't did that, its not getting used/referenced anywhere else

Reply:
It's necessary because of the following factors:
* Performance as constants are optimized by the compiler and runtime. That means, when running requests, it'll be able to access the field quicker, even if it's negligible for this string variable.
* Readability, if it's not declared a constant then anybody looking at the code will this is being modified.
* We also get a safety net because nobody will accidently modify the URL. 
* IDE support around static analysis. Intellij can better spot bugs or inconsistencies and making a variable that's not meant to be modified `final` will help with that.